apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java

```diff
@@ -0,0 +1,119 @@
+package gov.cms.bfd.pipeline.rda.grpc.server;
+
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.collect.ImmutableList;
+import gov.cms.mpsm.rda.v1.fiss.FissClaim;
+import gov.cms.mpsm.rda.v1.fiss.FissClaimStatus;
+import gov.cms.mpsm.rda.v1.fiss.FissProcedureCode;
+import gov.cms.mpsm.rda.v1.fiss.FissProcessingType;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * Objects of this class create populated FissClaim objects using random data. The purpose is simply
+ * to rapidly produce objects for pipeline testing to try out different scenarios for
+ * transformation. The purpose is NOT to produce realistic/valid data. The random number seed is
+ * settable in the constructor to allow for for predictable unit tests. Every optional field has a
+ * 50% chance of being present in each claim. Arrays have randomly assigned variable length
+ * (including zero).
+ */
+public class RandomFissClaimGenerator {
+  private static final EnumSet<FissClaimStatus> SKIPPED_CLAIM_STATUS_SET =
+      EnumSet.of(FissClaimStatus.UNRECOGNIZED, FissClaimStatus.CLAIM_STATUS_UNSET);
+  private static final EnumSet<FissProcessingType> SKIPPED_PROC_TYPE_SET =
+      EnumSet.of(FissProcessingType.UNRECOGNIZED, FissProcessingType.PROCESSING_TYPE_UNSET);
+  private static final String ALPHA = "bcdfghjkmnpqrstvwxz";
+  private static final String DIGIT = "1234567890";
+  private static final String ALNUM = ALPHA + DIGIT;
+  private static final int MAX_DAYS_AGO = 180;
+  private static final int MAX_PROC_CODES = 7;
+  private static final List<FissClaimStatus> CLAIM_STATUSES =
+      Arrays.stream(FissClaimStatus.values())
+          .filter(x -> !SKIPPED_CLAIM_STATUS_SET.contains(x))
+          .collect(Collectors.collectingAndThen(toList(), ImmutableList::copyOf));
+  private static final List<FissProcessingType> PROCESSING_TYPES =
+      Arrays.stream(FissProcessingType.values())
+          .filter(x -> !SKIPPED_PROC_TYPE_SET.contains(x))
+          .collect(Collectors.collectingAndThen(toList(), ImmutableList::copyOf));
+
+  private final Random random;
+
+  public RandomFissClaimGenerator(long seed) {
+    this.random = new Random(seed);
+  }
+
+  public FissClaim randomClaim() {
```

Comment:
This may go outside of the scope of this class, but one question we're interested in knowing is at what point does the current UPSERT logic start to degrade because the number of UPDATES is much higher than the number of INSERTS. If there was a way to control the percentage of uniquely generated IDs in a run, we may be able to test that with this class

Reply:
That's an excellent point.  I tested that out locally by adjusting the length of the DCN generated by `RandomFissClaimGenerator` (fewer digits mean more repeats) but that's not ideal at all.  An explicit setting as you propose would be easier to use.  Maybe we could add that when extending the server to also support MCS?

Reply:
yeah that sounds great.   with the current IT tests I'm less sure of its value, I'd rather see this setup with a postgres DB before we start drawing any conclusions.  so something to iterate on in the future